python/ray/serve/_private/replica.py

```diff
@@ -196,14 +198,17 @@ async def initialize_replica():
                     controller_handle,
                     app_name,
                 )
-                self._init_finish_event.set()
+                self._initialized = True
 
             # Is it fine that replica is None here?
             # Should we add a check in all methods that use self.replica
             # or, alternatively, create an async get_replica() method?
             self.replica = None
             self._initialize_replica = initialize_replica
 
+            # Used to make sure only one single init task happen.
+            self._replica_init_lock = asyncio.Lock()
```

Comment:
```suggestion
            # Used to guard `initialize_replica` so that it isn't called twice.
            self._replica_init_lock = asyncio.Lock()
```